pandas/core/arrays/base.py

```diff
@@ -1012,6 +1018,12 @@ def _pad_or_backfill(
                 DeprecationWarning,
                 stacklevel=find_stack_level(),
             )
+            if limit_area is not None:
+                raise NotImplementedError(
```

Comment:
We only get here if 3rd party authors didn't implement this themselves, correct?

Reply:
Right - this is only hit when EA author overrides `.fillna` but not `._pad_or_backfill`. Currently pandas `.ffill` will call the EA's `.fillna` in such a case, which can only be correctly done when `limit_area` is None.